bitnami/thanos/templates/query/deployment.yaml

```diff
@@ -88,9 +88,6 @@ spec:
             {{- if and .Values.receive.enabled $query.dnsDiscovery.enabled }}
             - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-receive{{ if .Values.receive.service.additionalHeadless }}-headless{{ end }}.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
             {{- end }}
-            {{- if and .Values.receive.enabled $query.dnsDiscovery.enabled }}
-            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-receive.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
-            {{- end }}
```

Comment:
Shouldn't we use this instead?
```yaml
            {{- if and .Values.receive.enabled .Values.receive.service.additionalHeadless $query.dnsDiscovery.enabled }}
            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-receive-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
            {{- end }}
            {{- if and .Values.receive.enabled $query.dnsDiscovery.enabled }}
            - --store=dnssrv+_grpc._tcp.{{ include "common.names.fullname" . }}-receive.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
            {{- end }}
```
This way you don't have the conflict, and you can use the flat both adding `-headless` and without it when `receive.service.additionalHeadless` is `true`.

Reply:
If we do this, we would have the same store twice on the query component.

Reply:
Thanos query would send every query to every store available, having the same store duplicated on the config doesn't help performance.
You should use one or the other.

Reply:
@juan131 the line 89 is creating receive and receive-headless

Reply:
No need for line 91 ongoing they got introduced because my rebase was not smooth and someone else added the receive discovery without headless support.

Reply:
@juan131 you don't want to use the normal service for discovery when headless is true

Reply:
because thanos is then not seeing the other receive components.

Reply:
@juan131 could we merge it?

Reply:
I want to have this change also in my PR #5504

Reply:
Or do I need to cherry-pick?

Reply:
@guilhermef could you please rebase?

Reply:
@juan131 I think it makes sense to drop the second line.

Reply:
Thanks for all the clarifications, i think the key is here:

>  you don't want to use the normal service for discovery when headless is true

So yes, it makes sense to drop the lines. 

@guilhermef please rebase the PR, and ignore my suggestion.